src/components/object/create/index.tsx

```diff
@@ -122,31 +118,46 @@ export const CreateObject = ({ appendLog }) => {
             }}
           />
           <br />
-          <h4 className="block font-medium text-gray-600">Input IPFS/Arweave links:</h4>
+          <h4 className="block font-medium text-gray-600">Input IPFS/Arweave image links:</h4>
           <textarea
             className=" h-[100px] overflow-y-auto p-4 bg-white text-gray-600 rounded-lg border border-gray-300 w-full md:w-1/2 lg:w-1/3"
-            value={linkInfo.links.join('\n')}
+            value={imageLinks.links.join('\n')}
             placeholder="links separated by newline"
             style={{ width: '100%', marginBottom: 5 }}
             onChange={(e) => {
               const newLinks = e.target.value.split('\n');
-              setLinkInfo({ ...linkInfo, links: newLinks });
+              setImageLinks({ ...imageLinks, links: newLinks });
+            }}
+          />
+          <br />
+          <h4 className="block font-medium text-gray-600">Metadata links (optional):</h4>
+          <textarea
+            className=" h-[100px] overflow-y-auto p-4 bg-white text-gray-600 rounded-lg border border-gray-300 w-full md:w-1/2 lg:w-1/3"
+            value={metadataLinks.links.join('\n')}
+            placeholder="links separated by newline"
+            style={{ width: '100%', marginBottom: 5 }}
+            onChange={(e) => {
+              const newLinks = e.target.value.split('\n');
+              setMetadataLinks({ ...metadataLinks, links: newLinks });
             }}
           />
           <br />
           <button className="bg-green-600 px-4 py-2 text-white hover:bg-green-500 sm:px-8 sm:py-3 rounded-lg w-full" style={{ marginBottom: 5 }}
             onClick={async () => {
               appendLog('Initializing...');
-              if (!linkInfo || !linkInfo.links.length) {
-                appendLog('Please set links');
+              if (!imageLinks || !imageLinks.links.length) {
+                appendLog('Please set image links');
                 return;
               }
               if (!address) {
                 appendLog('Please select an address');
                 return;
               }
-              for (const singleLink of linkInfo.links) {
-                await uploadFiles(singleLink, createObjectInfo, appendLog, setProgress, connector, address, chain);
+              for (const singleImageLink of imageLinks.links) {
+                await uploadFiles(singleImageLink, UploadType.Image, createObjectInfo, appendLog, setProgress, connector, address, chain);
+              }
+              for (const singleMetadataLink of metadataLinks.links) {
+                await uploadFiles(singleMetadataLink, UploadType.Metadata, createObjectInfo, appendLog, setProgress, connector, address, chain);
```

Comment:
metadata should be updated with new link to image in greenfield

Reply:
Fixed, however we need to also figure out which link to share